src/entity.ts

```diff
@@ -52,27 +52,44 @@ export namespace entity {
    * provided.
    *
    * @class
-   * @param {number} value The double value.
+   * @param {number|string} value The double value.
    *
    * @example
    * const {Datastore} = require('@google-cloud/datastore');
    * const datastore = new Datastore();
    * const aDouble = datastore.double(7.3);
    */
-  export class Double {
+  export class Double extends Number {
+    private _entityPropertyName: string | undefined;
     type: string;
     value: number;
-    constructor(value: number) {
+    constructor(value: number | string | ValueProto) {
+      super(typeof value === 'object' ? value.doubleValue : value);
+
       /**
        * @name Double#type
        * @type {string}
        */
       this.type = 'DatastoreDouble';
+
+      this._entityPropertyName =
+        typeof value === 'object' ? value.propertyName : undefined;
+
       /**
        * @name Double#value
        * @type {number}
        */
-      this.value = value;
+      this.value =
+        typeof value === 'object' ? Number(value.doubleValue) : Number(value);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    valueOf(): any {
+      return Number(this.value);
```

Comment:
Playing around with extending `Number`, there isn't access to a `this.value` unless we store the value ourselves:

```js
class F extends Number {
  constructor(value) {
    super(value)
  }
  blerg() {
    console.info(this.value)
  }
}

const f = new F(99);
console.info(f.valueOf())
f.blerg()
```

outputs:

```
99
undefined
```

~Part of me wonders if we should avoid extending the base `Number` type, and instead create a `DatastoreNumber` type, since `Number` implies a a larger interface than we potentially want to support out of the gate?~

Edit: I see we're already extending `Number` in `Int`, I'd stick with this approach.

Reply:
```typescript
class F extends Number {
  constructor(value) {
    super(value);
  }
  blerg() {
    console.info(Number(this));
  }
}

const f = new F(99);
f.blerg(); // outputs 99 as expected
```